### PR TITLE
Fix WordPress fatal error when plugin is activated before creating a network (WP-625)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**
Fix in 2ad14c11085664948771898f5c84d11dcafb66c4, other two commits are refactoring and version bump

### CODE REVIEW

File: inc/Smartling/Extensions/Acf/AcfDynamicSupport.php [hunk 1]
```diff
@@ -473,7 +425,14 @@ private function tryRegisterACF()
                 DiagnosticsHelper::addDiagnosticsMessage(implode('<br/>', $msg));
                 $this->getLogger()->notice('Automatic ACF support is disabled.');
             } else {
-                $dbDefinitions = $this->getDatabaseDefinitions();
+                try {
+                    $dbDefinitions = $this->getDatabaseDefinitions();
+                } catch (SmartlingDirectRunRuntimeException $e) {
+                    $dbDefinitions = [];
+                    DiagnosticsHelper::addDiagnosticsMessage(
+                        'Failed to get ACF definitions from database. Please ensure that WordPress network is set up properly'
```
Comment (PavelLoparev) [inline inc/Smartling/Extensions/Acf/AcfDynamicSupport.php:433]: Maybe let's print the exception message for more details?

File: readme.txt [hunk 2]
```diff
@@ -60,13 +60,16 @@ Additional information on the Smartling Connector for WordPress can be found [he
 3. Track translation status within WordPress from the Submissions Board. View overall progress of submitted translation requests as well as resend updated content.
 
 == Changelog ==
-= 2.4.14
+= 2.4.15 =
+* Fix fatal error when plugin is activated before creating a network
```
Comment (PavelLoparev) [inline readme.txt:64]: `Fixed` or `Fix for`

File: inc/Smartling/Extensions/Acf/AcfDynamicSupport.php (hunk 1 above)
Comment (dimitrystd) [reply inc/Smartling/Extensions/Acf/AcfDynamicSupport.php:433]: +1 to add the original $e message as the second line

**Review by dimitrystd — COMMENTED** [review]
+1 for refactoring but I didn't get the fix. It seems just try-catch was added.

File: inc/Smartling/Extensions/Acf/AcfDynamicSupport.php (hunk 1 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Extensions/Acf/AcfDynamicSupport.php:433]: Improved

File: readme.txt (hunk 2 above)
Comment (vsolovei-smartling) [reply readme.txt]: Log entry removed, won't be in a release of it's own

**Comment by vsolovei-smartling** [comment]
> +1 for refactoring but I didn't get the fix. It seems just try-catch was added.

This is the fix: without it the behaviour is to throw exception, doesn't get caught, user unable to use admin panels.

**Review by PavelLoparev — APPROVED** [review]

**Review by dimitrystd — APPROVED** [review]
It seems you have to pull master into the branch before merge it.

**Pull request merged**